rust-runtime/aws-smithy-types/src/timeout.rs

```diff
@@ -0,0 +1,261 @@
+    pub fn connect_timeout(&self) -> Option<Duration> {
+        self.connect_timeout
+    }
+
+    /// A limit on the amount of time a TLS handshake takes from when the `CLIENT HELLO` message is
+    /// sent to the time the client and server have fully negotiated ciphers and exchanged keys.
+    pub fn tls_negotiation_timeout(&self) -> Option<Duration> {
+        self.tls_negotiation_timeout
+    }
+
+    /// A limit on the amount of time an application takes to attempt to read the first byte over an
+    /// established, open connection after write request. This is also known as the
+    /// "time to first byte" timeout.
+    pub fn read_timeout(&self) -> Option<Duration> {
+        self.read_timeout
+    }
+
+    /// A limit on the amount of time it takes for the first byte to be sent over an established,
+    /// open connection and when the last byte is received from the service for a single attempt.
+    /// If you want to set a timeout for an entire request including retry attempts,
+    /// use [`TimeoutConfig::api_call_timeout`] instead.
+    pub fn api_call_attempt_timeout(&self) -> Option<Duration> {
+        self.api_call_attempt_timeout
+    }
+
+    /// A limit on the amount of time it takes for request to complete. A single request may be
+    /// comprised of several attemps depending on an app's [`RetryConfig`](super::retry::RetryConfig). If you want
+    /// to control timeouts for a single attempt, use [`TimeoutConfig::api_call_attempt_timeout`].
+    pub fn api_call_timeout(&self) -> Option<Duration> {
+        self.api_call_timeout
+    }
+
+    /// Consume a `TimeoutConfig` to create a new one, setting the connect timeout
+    pub fn with_connect_timeout(mut self, timeout: Option<Duration>) -> Self {
+        self.connect_timeout = timeout;
+        self
+    }
+
+    /// Consume a `TimeoutConfig` to create a new one, setting the TLS negotiation timeout
+    pub fn with_tls_negotiation_timeout(mut self, timeout: Option<Duration>) -> Self {
+        self.tls_negotiation_timeout = timeout;
+        self
+    }
+
+    /// Consume a `TimeoutConfig` to create a new one, setting the read timeout
+    pub fn with_read_timeout(mut self, timeout: Option<Duration>) -> Self {
+        self.read_timeout = timeout;
+        self
+    }
+
+    /// Consume a `TimeoutConfig` to create a new one, setting the API call attempt timeout
+    pub fn with_api_call_attempt_timeout(mut self, timeout: Option<Duration>) -> Self {
+        self.api_call_attempt_timeout = timeout;
+        self
+    }
+
+    /// Consume a `TimeoutConfig` to create a new one, setting the API call timeout
+    pub fn with_api_call_timeout(mut self, timeout: Option<Duration>) -> Self {
+        self.api_call_timeout = timeout;
+        self
+    }
+
+    /// Merge two builders together. Values from `other` will only be used as a fallback for values
```

Comment:
Rust docs should generally have a short summary line that stands alone for the abridged summary that shows up in the generated docs.
```suggestion
    /// Merges two builders together.
    ///
    /// Values from `other` will only be used as a fallback for values
```